packages/opencensus-instrumentation-redis/test/test-redis.ts

```diff
@@ -0,0 +1,217 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CoreTracer, logger, RootSpan, Span, SpanEventListener, SpanKind} from '@opencensus/core';
+import * as assert from 'assert';
+import * as redis from 'redis';
+
+import {plugin} from '../src';
+
+/** Collects ended root spans to allow for later analysis. */
+class RootSpanVerifier implements SpanEventListener {
+  endedRootSpans: RootSpan[] = [];
+
+  onStartSpan(span: RootSpan): void {
+    return;
+  }
+  onEndSpan(root: RootSpan) {
+    this.endedRootSpans.push(root);
+  }
+}
+
+/**
+ * Asserts root spans attributes.
+ * @param rootSpanVerifier An instance of rootSpanVerifier to analyse RootSpan
+ * instances from.
+ * @param expectedName The expected name of the first root span.
+ * @param expectedKind The expected kind of the first root span.
+ */
+function assertSpan(
+    rootSpanVerifier: RootSpanVerifier, expectedName: string,
+    expectedKind: SpanKind, verifyAttribute?: (span: Span) => boolean) {
+  assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 1);
+  assert.strictEqual(rootSpanVerifier.endedRootSpans[0].spans.length, 1);
+  assert.strictEqual(
+      rootSpanVerifier.endedRootSpans[0].spans[0].name, expectedName);
+  assert.strictEqual(
+      rootSpanVerifier.endedRootSpans[0].spans[0].kind, expectedKind);
+  if (typeof verifyAttribute === 'function') {
+    for (const span of rootSpanVerifier.endedRootSpans[0].spans) {
+      assert(verifyAttribute(span), 'failed to verify attribute');
+    }
+  }
+}
+
+describe('RedisPlugin', () => {
+  // For these tests, mongo must be runing. Add OPENCENSUS_REDIS_TESTS to run
+  // these tests.
+  const OPENCENSUS_REDIS_TESTS = process.env.OPENCENSUS_REDIS_TESTS as string;
+  const OPENCENSUS_REDIS_HOST = process.env.OPENCENSUS_REDIS_HOST as string;
+  let shouldTest = true;
+  if (!OPENCENSUS_REDIS_TESTS) {
+    console.log('Skipping test-redis. Run REDIS to test');
+    shouldTest = false;
+  }
+
+  const URL = `redis://${OPENCENSUS_REDIS_HOST || 'localhost'}:6379`;
+  const VERSION = '2.8.0';
+  const REDIS_QUERY_TYPE = SpanKind.CLIENT;
+
+  const tracer = new CoreTracer();
+  const rootSpanVerifier = new RootSpanVerifier();
+  let client: redis.RedisClient;
+
+  before((done) => {
+    tracer.start({samplingRate: 1, logger: logger.logger(1)});
+    tracer.registerSpanEventListener(rootSpanVerifier);
+    plugin.enable(redis, tracer, VERSION, {}, '');
+    client = redis.createClient({url: URL});
+    client.on('error', (err: Error) => {
+      console.log(
+          'Skipping test-redis. Could not connect. Run Redis to test', err);
```

Comment:
If the user selected that the Redis test should run via the environment config, then should this just fail rather than passing with a log message?

Reply:
It's the current behavior of the mongodb test, i agree with you that it should not continue. Maybe could we change this behavior in another PR ?

Reply:
Hmm, I guess I'd be inclined to have it fail here for the new Redis instrumentation since this is new code and then create an issue or follow up PR to change the Mongo instrumentation to match.

But @mayurkale22 what do you think? I'd leave this decision up to you.

Reply:
I second what @draffensperger mentioned. We should make change here and open an issue for mongodb plugin.

Reply:
done